src/models/train.py

```diff
@@ -90,7 +92,7 @@ def instantiate_trainer(self, **kwargs: Any) -> Trainer:
 
         if self.datamodule.val_path:
             if self.datamodule.is_classification:
-                checkpoint_metric = "val/roc_auc"
+                checkpoint_metric = "val/{}".format(self.config["fit"]["optimize"])
                 mode = "max"
             else:
                 checkpoint_metric = "val/loss"
```

Comment:
Feels a little weird to me that the argument gets ignored if it's a regression task. Maybe the default value in the parser is `None`, and if that value is passed we use `val/roc_auc` or `train/loss`.